src/Evolution/DiscontinuousGalerkin/DgElementArray.hpp

```diff
@@ -110,33 +112,26 @@ void DgElementArray<Metavariables, PhaseDepActionList>::allocate_array(
       get<domain::Tags::InitialExtents<volume_dim>>(initialization_items);
   const auto& quadrature =
       get<evolution::dg::Tags::Quadrature>(initialization_items);
-
-  bool use_z_order_distribution = true;
-  if constexpr (detail::has_use_z_order_distribution_v<Metavariables>) {
-    use_z_order_distribution = Metavariables::use_z_order_distribution;
-  }
-
-  bool local_time_stepping = false;
-  if constexpr (detail::has_local_time_stepping_v<Metavariables>) {
-    local_time_stepping = Metavariables::local_time_stepping;
-  }
+  const std::optional<domain::ElementWeight>& element_weight =
+      Parallel::get<domain::Tags::ElementDistribution>(local_cache);
 
   const size_t number_of_procs = Parallel::number_of_procs<size_t>(local_cache);
   const size_t number_of_nodes = Parallel::number_of_nodes<size_t>(local_cache);
   const size_t num_of_procs_to_use = number_of_procs - procs_to_ignore.size();
 
   const auto& blocks = domain.blocks();
 
-  const std::unordered_map<ElementId<volume_dim>, double> element_costs =
-      domain::get_element_costs(
-          blocks, initial_refinement_levels, initial_extents,
-          local_time_stepping
-              ? domain::ElementWeight::NumGridPointsAndGridSpacing
-              : domain::ElementWeight::NumGridPoints,
-          quadrature);
-  const domain::BlockZCurveProcDistribution<volume_dim> element_distribution{
-      element_costs,   num_of_procs_to_use, blocks, initial_refinement_levels,
-      initial_extents, procs_to_ignore};
+  // Only need this if the element weight has a value
```

Comment:
`this` -> `element_distribution` (so that when someone accidentally moves the comment or code, at least we have some hope of putting them back together :D )

Maybe expand comment `... since then we have to distribute the elements in a more complicated way than just round robin`?